mesonbuild/compilers/mixins/clike.py

```diff
@@ -476,6 +476,17 @@ def cross_compute_int(self, expression: str, low: T.Optional[int], high: T.Optio
             if self._compile_int(f'{expression} == {guess}', prefix, env, extra_args, dependencies):
                 return guess
 
+        # Try to expand the expresseion and evaluate it on the build machines compiler
+        expanded, _ = self.get_define(expression, prefix, env, extra_args, dependencies, False)
+        evaluate_expanded = f'''
+        #include <stdio.h>
+        #include <stdint.h>
+        int main(void) {{ int expression = {expanded}; printf("%d", expression); return 0; }}'''
+        run = env.coredata.compilers[mesonlib.MachineChoice.BUILD]['c'].run(evaluate_expanded, env)
+        if run and run.compiled and run.returncode == 0:
+            if self._compile_int(f'{expression} == {run.stdout}', prefix, env, extra_args, dependencies):
+                return int(run.stdout)
```

Comment:
Ahh, I think this isn't going to give correct results all the time. For example, if the expression contains `long` in it, and you are cross-compiling to win32.

Reply:
Well, the idea is that `run` equals the potentially incorrect answer and we use `run.stdout` as one of our guesses, right? We're still only returning from the function if `self._compile_int()` also agrees the answer is correct, and that is a true cross check usually used to do a full binary search.

So this evaluation attempt may be useless but it shouldn't be able to return incorrect results. The question is whether it can ever (more importantly, "often enough") guess the correct answer faster than repeatedly guessing numbers in the bisection of minint ----> maxint.

Reply:
Ah it's returning a guess. I missed (forgot) that. LGTM then.